pyproject.toml

```
@@ -46,6 +46,7 @@ Issues = "https://github.com/secure-systems-lab/securesystemslib/issues"
 crypto = ["cryptography>=37.0.0"]
 pynacl = ["pynacl>1.2.0"]
 PySPX = ["PySPX==0.5.0"]
+hsm = ["asn1crypto", "cryptography", "PyKCS11"]
```

Should this be: 
```suggestion
asn1crypto = ["asn1crypto"]
PyKCS11 = ["PyKCS11"]
```
?

For the sake of consistency, it should. Also,  specifying optional dependencies per feature is prone to redundancy (see cryptography).

Either way, we should document which optional dependency is needed for which currently available feature.